### ISSUE TITLE
mDNS resolver not discovering IPv6 temporary addresses

### ISSUE THREAD

**Issue by Fail-Safe** [nextdns/nextdns#39]
I have been trying to troubleshoot why my NextDNS logs do not show a proper device name for IPv6 capable clients which also have temporary addresses. I now see what is causing the discrepancy.

Here's an example of one of my clients (partially obscured IPs to protect the innocent):
```
I: 15:43:06 Discovered 192.168.XX.145 = MBP-2
I: 15:43:06 Discovered fe80::c1a:aaaa:a1b2:f7b7 = MBP-2
I: 15:43:06 Discovered 2606:XXXX:YYYY:ZZZZ:4d2:9953:51c2:8583 = MBP-2
```

Here is the ifconfig info from this same client:
```
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	inet6 fe80::c1a:aaaa:a1b2:f7b7%en0 prefixlen 64 secured scopeid 0x4
	inet 192.168.XX.145 netmask 0xffffff00 broadcast 192.168.45.255
	inet6 2606:XXXX:YYYY:ZZZZ:4d2:9953:51c2:8583 prefixlen 64 autoconf secured
	inet6 2606:XXXX:YYYY:ZZZZ:81c:58e3:329b:690f prefixlen 64 autoconf temporary
```

So the mDNS resolver is only discovering the IPv4, IPv6-LL, and IPv6-secured addresses. But due to the nature of IPv6 when temporary addressing is used, outbound connections usually use the temporary address. In this case, the temporary address is what is getting ushered to my NextDNS client (via dnsmasq `add-subnet=32,128`) and subsequently logged with the query. The result is a device name such as `Device #7DSZQ` instead of `MBP-2` (in this example) on the NextDNS side.

Is there a way for mDNS resolver to also discover or otherwise account for IPv6 temporary addresses?

**Comment by rs** [comment]
We need to investigate.

**Comment by rs** [comment]
Please test with master

**Comment by Fail-Safe** [comment]
@rs Could this be revisited? I am still seeing issues with detection of IPv6 temporary addresses. I added more info on your commit here: https://github.com/nextdns/nextdns/commit/29b36779b4ca0a02cbe15627a252374f8d3d1c34

**Comment by rs** [comment]
I will add ndp support

**Comment by Fail-Safe** [comment]
Sounds good. I sure wish I could help out with that, but I'm not very Go proficient at this point. I'm a Python guy these days.

**Comment by Fail-Safe** [comment]
FWIW, I did more testing on a couple of different scenarios. I'm not sure what to make of it at the moment--has me stumped.

Keeping in mind all of this testing is with OpenWRT and the 1.5.8 dev build (rebuilt last night after your latest commits). Dnsmasq in all configurations has cache disabled. Caching is only set on NextDNS client.

Combinations that do not work for SLAAC temporary address resolution:

1. NextDNS client --> listening port `:53` and split-horizon to dnsmasq (via `127.0.0.1:5053`) for local domain resolution
dnsmasq           --> listening port `:5053` bound to each local interface

2. NextDNS client --> listening port `127.0.0.1:5053`
dnsmasq --> listening port `:53` bound to each local interface and forwarding to NextDNS client (via `127.0.0.1:5053`)

However, in this configuration, I have successful SLAAC temporary address resolution, thus my NextDNS console shows all device names properly:

3. NextDNS client --> listening port `:5053` and split-horizon to dnsmasq (via `127.0.0.1:53`) for local domain resolution
dnsmasq --> listening port `:53` bound to each local interface and forwarding to NextDNS client (via `127.0.0.1:5053`)

But, even in the working configuration 3, `nextdns discovered` output does not indicate any record of the temporary IPv6 addresses that got logged in the console. Further, with the `ip6neigh` process (which I referred to yesterday) disabled, dnsmasq does not resolve my AAAA local records. So I have no clue where they are getting resolved, but they are.

I would keep this working configuration and not "complain" (I'm truly not complaining--the NextDNS client is awesome), but the NextDNS client keeps writing out `/tmp/dnsmasq.d/nextdns.conf as the following which breaks dnsmasq on reboot/service restart:
```
server=#5053
no-resolv
add-mac
add-subnet=32,128
```

Anyway, not sure if any of this is useful information. But it sure has me puzzled at the moment.
